### PR TITLE
Migrate to gspell

### LINKED ISSUE #93: Port to GSpell
Seems that would be pretty adequate to transition from GtkSpellCheck to Gspell at some point

### PR DESCRIPTION
**Pull request by BrainBlasted**
Closes #93. Could not test due to #149. Flatpak
changes are based on #150

### CODE REVIEW

**Comment by goncalossilva** [comment]
This is great. :raised_hands: :raised_hands: 

I've tested this and there are a couple differences in UX:
* The underline style is different ([old](https://cl.ly/7dd4b86eeaf9/gnome-shell-screenshot-EQGM0Z.png) vs [new](https://cl.ly/0ecd140dd38e/gnome-shell-screenshot-GXGO0Z.png), I personally prefer the new one)
* During focus mode, "unfocused" lines were not spell-checked, now they are, due to [this](https://github.com/UberWriter/uberwriter/pull/151/files#diff-35ffc34a91c5e171a00d023a9175ab9eL227) removal. Is there a way to limit GSpell to specific lines in this case?

There is also a warning on startup:
```
/home/goncalossilva/Code/Linux/uberwriter/uberwriter/text_view.py:11: PyGIWarning: Gspell was imported without specifying a version first. Use gi.require_version('Gspell', '1') before import to ensure that the right version gets loaded.
  from gi.repository import Gtk, Gdk, GObject, GLib, Gspell
```

Otherwise it works very well for me locally. :+1:

**Comment by BrainBlasted** [comment]
I just pushed a change that fixed both issues - can you try it out?

I found that on stable spellchecking was just off all-together in `focus_mode`. I decided to replicate that behavior since Gspell can't set spellcheck for specific regions.

**Comment by goncalossilva** [comment]
It needs to be Gspell, not GSpell :smile: 

```
Traceback (most recent call last):
  File "./uberwriter.in", line 49, in <module>
    import uberwriter
  File "/home/goncalossilva/Code/Linux/uberwriter/uberwriter/__init__.py", line 22, in <module>
    from uberwriter import window
  File "/home/goncalossilva/Code/Linux/uberwriter/uberwriter/window.py", line 29, in <module>
    from uberwriter.text_view import TextView
  File "/home/goncalossilva/Code/Linux/uberwriter/uberwriter/text_view.py", line 11, in <module>
    gi.require_version('GSpell', '1')
  File "/usr/lib64/python3.7/site-packages/gi/__init__.py", line 129, in require_version
    raise ValueError('Namespace %s not available' % namespace)
ValueError: Namespace GSpell not available
```

Changing to `gi.require_version('Gspell', '1')` addresses it.

There is one edge case:
1. Enter focus mode (spell-checking is disabled)
2. Go to the settings and disable->enable spell-checking
3. App is still in focus mode, but spell checking is enabled

To account for both, how about having a property/method on the text_view, eg. `set_spellcheck_enabled`, and then use it like so `self.gspell_view.set_inline_spell_checking(self.spellcheck_enabled and not self.focus_mode)`?

**Comment by BrainBlasted** [comment]
> It needs to be Gspell, not GSpell

Oops 🤦🏾‍♂️ Fixed.

> To account for both, how about having a property/method on the text_view, eg. set_spellcheck_enabled, and then use it like so?

Alright, changed in window.py's `toggle_spellcheck()` like so:

```python3
        self.text_view.gspell_view\
            .set_inline_spell_checking(state.get_boolean()
                                       and not self.text_view.focus_mode)
```

**Comment by goncalossilva** [comment]
Works great now. :+1:

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
 regex
 enchant
-python-gtkspellcheck
+gspell
```
Comment (goncalossilva) [inline requirements.txt]: Isn't this a system dependency? pip doesn't find this.

Comment (BrainBlasted) [reply requirements.txt]: Ah, I didn't know that was only for pip deps. Just assumed it was a list of general dependencies.

**Comment by BrainBlasted** [comment]
> auto_correct.py still uses enchant in a few places. I don't know the background behind auto-correct, but we should either update it as well, or get rid of it considering that it's currently unused.

I assumed that was something to be implemented later and didn't fit within the scope of the PR, so I didn't touch it. @somas95 do you still plan on implementing auto-correct? If not I can remove it.

**Comment by somas95** [comment]
merged 6c50e53c9dce62f8a578454996b506e0d8bd87fb

**Comment by goncalossilva** [comment]
@somas95 what to do with `auto_correct.py`? Leave it be?

**Comment by somas95** [comment]
I would leave it there for the moment (it's not called in any place)
I'll work on it in a future if I can

**Comment by hThoreau** [comment]
Hey, how do you add languages to the spellchecker? Is it possible for the fltapak package?

**Pull request closed without merging**